src/useFirebaseUser.js

```diff
@@ -72,9 +72,9 @@ const useFirebaseUser = () => {
       const claims = filterStandardClaims(idTokenResult.claims)
       setCustomClaims(claims)
     }
+    await setAuthCookie(firebaseUser)
```

Comment:
This would fix the issue but if you take a closer look at @kmjennison's comment (https://github.com/gladly-team/next-firebase-auth/issues/189#issuecomment-854220029), he actually suggests a different approach so we don't make this blocking.

> However, that would require the app to wait for a server call every time before the auth is initialized, which would be a bad performance hit.

> A better approach would be to return another property from useFirebaseUser, like setAuthCookieComplete, then wait for that to be true before redirecting to the app. This would wait for the server call only on pages that should redirect when the user is authenticated. Note that the client won't know for sure if the cookies were in fact set, given they're HTTP-only.

Reply:
+1 – for performance, please take the second approach.

Reply:
I'm not sure if I followed, but I updated the PR and now `useFirebaseUser` returns a new boolean property named `authCookieComplete` that is set to true asynchronously after `setAuthCookie` successfully runs.

Reply:
@moises-marquez Just wanted to let you know this is still on my radar. Apologies for the delay in reviewing, it's been a busy few weeks.

Reply:
No worries. It's summertime and people are usually having a good time ;).